src/oss/langchain/middleware.mdx

```diff
@@ -453,20 +456,112 @@ const agent = createAgent({
 ```
 :::
 
+### Model fallback
+
+The `ModelFallbackMiddleware` provides automatic model fallback on errors. This middleware attempts to retry failed model calls with alternative models in sequence. When a model call fails, it tries the next model in the fallback list until either a call succeeds or all models have been exhausted.
+
+**Key features:**
+
+- Automatic retry with fallback models when primary model fails
+- Sequential fallback through multiple models
+- Preserves original request parameters while switching models
+- Configurable with any combination of model strings or instances
+
+**Use Cases:**
+
+- Handling model outages or rate limits
+- Cost optimization by trying cheaper models first
+- Ensuring high availability for critical applications
+
+:::python
+```python
+from langchain.agents import create_agent
+from langchain.agents.middleware import ModelFallbackMiddleware
+
+agent = create_agent(
+    model="openai:gpt-4o",  # Primary model
+    tools=[weather_tool, calculator_tool],
+    middleware=[
+        ModelFallbackMiddleware(
+            "openai:gpt-4o-mini",  # First fallback
+            "anthropic:claude-3-5-sonnet-20241022",  # Second fallback
+        ),
+    ],
+)
+
+# If gpt-4o fails, automatically tries gpt-4o-mini, then claude
+result = agent.invoke({"messages": [HumanMessage("Hello")]})
+```
+:::
+
+:::js
+```typescript
+import { createAgent, modelFallbackMiddleware, HumanMessage } from "langchain";
+
+const agent = createAgent({
+  model: "openai:gpt-4o", // Primary model
+  tools: [weatherTool, calculatorTool],
+  middleware: [
+    modelFallbackMiddleware(
+      "openai:gpt-4o-mini", // First fallback
+      "anthropic:claude-3-5-sonnet-20241022" // Second fallback
+    ),
+  ],
+});
+
+// If gpt-4o fails, automatically tries gpt-4o-mini, then claude
+const result = await agent.invoke({
+  messages: [new HumanMessage("Hello")]
+});
+```
+:::
+
+**Configuration:**
+
+:::python
+The `ModelFallbackMiddleware` constructor accepts fallback models in order of preference:
+
+- `first_model`: The first fallback model (required)
+- `*additional_models`: Additional fallback models in order
+
+Models can be specified as:
+- Model name strings (e.g., `"openai:gpt-4o-mini"`)
+- `BaseChatModel` instances for pre-configured models
+:::
+
+:::js
+The `modelFallbackMiddleware` function accepts fallback models in order of preference:
+
+- `...fallbackModels`: Fallback models in order of preference
+
+Models can be specified as:
+- Model name strings (e.g., `"openai:gpt-4o-mini"`)
+- `LanguageModelLike` instances for pre-configured models
+:::
+
+The middleware works by:
```

Comment:
I think we can remove this? It's similar info to the opening sentences at the top of the section